File: web/src/types/threat.ts
export type Severity = "LOW" | "MEDIUM" | "HIGH" | "CRITICAL";
export type ThreatStatus = "PENDING" | "VERIFIED" | "SETTLED";

export interface Threat {
  id: string;
  hash: string;
  severity: Severity;
  aiScore: number;
  reporterPubkey: string;
  validatorCount: number;
  status: ThreatStatus;
  timestamp: number;
  description: string;
  url: string;
}

File: programs/shieldmesh/src/state/threat.rs
use anchor_lang::prelude::*;

#[account]
#[derive(InitSpace)]
pub struct ThreatAccount {
    pub reporter: Pubkey,
    pub threat_hash: [u8; 32],
    pub severity: u8,
    pub ai_score: u8,
    pub validator_count: u8,
    /// 0 = PENDING, 1 = VERIFIED, 2 = SETTLED
    pub status: u8,
    pub timestamp: i64,
    pub bump: u8,
}
